src/lib/io/sol-iio.c

```diff
@@ -610,6 +628,11 @@ channel_get_pure_name(struct sol_iio_channel *channel)
         }
     }
 
+    if (modified) {
+        channel_pure_name = strndup(channel_name, channel_name_len);
```

Comment:
`strndup` may fail. I know that there's a non-verified call above. It should be fixed too. It maybe on another patch, though.